DEF[PORTVERSION]=	1.11.1
# ------------------------------------------------------------------------

NAMEBASE=		python-pygit2
VERSION=		${PORTVERSION}
KEYWORDS=		python
VARIANTS=		py310 v11
SDESC[py310]=		Python bindings for libgit2 (3.10)
SDESC[v11]=		Python bindings for libgit2 (3.11)
HOMEPAGE=		https://github.com/libgit2/pygit2
CONTACT=		Python_Automaton[python@ironwolf.systems]

DOWNLOAD_GROUPS=	main
SITES[main]=		PYPI/p/pygit2
DISTFILE[1]=		pygit2-${PORTVERSION}.tar.gz:main

SPKGS[py310]=		single
SPKGS[v11]=		single

OPTIONS_AVAILABLE=	PY310 PY311
OPTIONS_STANDARD=	none
VOPTS[py310]=		PY310=ON PY311=OFF
VOPTS[v11]=		PY310=OFF PY311=ON

# License listed at PyPI
# GPLv2 with linking exception

GENERATED=		yes
DISTNAME=		pygit2-${PORTVERSION}

# install_requires extracted from setup.py
# cffi>=1.9.1
# cffi>=1.9.1

[PY310].USES_ON=			python:py310,sutools
[PY310].BUILDRUN_DEPENDS_ON=		python-cffi:single:py310
[PY311].USES_ON=			python:v11,sutools
[PY311].BUILDRUN_DEPENDS_ON=		python-cffi:single:v11

BUILDRUN_DEPENDS=	libgit2:single:standard
MAKE_ENV=		LIBGIT2={{LOCALBASE}}

post-patch:
	# This version check is always too conservative, causing breakage
	${REINPLACE_CMD} -e '/#error You need a compatible libgit2/d' \
		${WRKSRC}/src/types.h
